include/odp/api/spec/packet.h

```diff
@@ -1120,63 +1156,121 @@ odp_packet_t odp_packet_reassemble(odp_pool_t pool, odp_packet_buf_t pkt_buf[],
- * has metadata initialized to default values and does not contain unshared
- * data.  Packet (head) manipulation functions may be used normally to, e.g.,
- * add a unique header onto the shared payload. The shared part of the packet
- * may be modified again when there is a single reference left. Static and
- * dynamic references must not be mixed. Results are undefined if these
- * restrictions are not observed.
- *
- * The packet handle 'pkt' may itself be a (dynamic) reference to a packet.
- *
- * If the caller does not intend to modify either the packet or the new
- * reference to it, odp_packet_ref_static() may be used to create
- * a static reference that is more optimized for that use case.
- *
- * Packet is not modified on failure.
- *
- * @param pkt    Handle of the packet for which a reference is to be
- *               created.
- *
- * @param offset Byte offset in the packet at which the shared part is to
- *               begin. This must be in the range 0 ... odp_packet_len(pkt)-1.
- *
- * @return New reference to the packet
+ * Create a packet that references packet data of another packet
+ *
+ * Return a packet that shares packet data with the given packet. 'pkt'
+ * becomes a 'referenced packet' and the returned packet will be a
+ * 'referencing packet'. The referencing packet is allocated from the
+ * same pool as the referenced packet.
+ *
+ * Packet metadata is not shared. The new packet has its metadata initialized
+ * to the default values.
+ *
+ * Shared packet data starts at the given offset of the referenced packet
+ * and continues to the end of the referenced packet. The referencing packet
+ * may also contain private, non-shared packet data after it has been modified
+ * using packet manipulation operation such as push and extend (see below).
+ * The referencing packet may also be modified to not actually contain any
+ * packet data of the referenced packet.
+ *
+ * Packet data and data layout of the referenced packet must not be modified
+ * as long as any packet that references the packet exists. In particular,
+ * a referenced packet must not be passed to odp_packet_{push,pull,extend,
+ * trunc}_{head,tail}(), odp_packet_split(), or as the destination packet to
+ * odp_packet_concat() or to any other ODP API function that could modify
+ * packet data or packet data layout.
+ *
+ * odp_packet_has_ref() returns 1 for the referenced packet as long as it is
+ * being referenced by other packets and 0 for the newly created referencing
+ * packet. odp_packet_is_referencing() returns 1 for the created referencing
+ * packet.
+ *
+ * The referenced and referencing packets can be freed in any order.
+ * ODP ensures that the packets or packet segments backing the shared packet
+ * data is put back to the packet pool only after there are no references left.
+ * An ODP implementation may implement this using per-packet or per-segment
+ * reference counting or by other means. See odp_packet_free().
```

Comment:
Unnecessary "An ODP implementation may implement this using per-packet or per-segment reference counting or by other means."?